### ISSUE TITLE
[Feature] Revert Filter UI or add a toggle

### ISSUE THREAD

**Issue by SpedNSFW** [stashapp/stash#3630]
**Is your feature request related to a problem? Please describe.**
The new filter UI is clunky and cumbersome. It also introduces some horrible UX, such as the performer gender filter; it uses checkboxes which implies you can select multiple, however you can only select one at a time (which in itself is horrible but not the point), switch anything like this to using radio fields as that is the entire purpose for them.

When the filter options were in a select element, you could display more of them at once, you could arrow-key through them, and you could literally type out the filter to jump directly to it.

**Describe the solution you'd like**
Either revert the filter UI or add a toggle in the settings so we can keep the previous way. The whole change made the UX worse overall for the sake of looking good. Function should _always_ come before form.

**Comment by WithoutPants** [comment]
> It also introduces some horrible UX, such as the performer gender filter; it uses checkboxes which implies you can select multiple, however you can only select one at a time (which in itself is horrible but not the point), switch anything like this to using radio fields as that is the entire purpose for them.

I agree that the Gender filter should not use checkboxes. It was changed from a dropdown to what it is now to save a click. I can change it to a radio until better gender filtering is introduced.

> When the filter options were in a select element, you could display more of them at once, you could arrow-key through them, and you could literally type out the filter to jump directly to it.

I am reasonably sure that with some custom css, you can fit as many options as you desire on the screen, or alternatively suggest a style that you think improves/maintains the real estate to allot to the headers. The others I agree are functional regressions on the previous behaviour and should be addressed.

> Either revert the filter UI or add a toggle in the settings so we can keep the previous way. 

You've listed two main issues here: checkboxes of options where radio should be, and being able to quickly navigate filter types with the keyboard. Instead of highlighting these as areas to fix, you suggest reverting the entire UI rework, or providing a way to use the old implementation, and then maintain the two UIs in perpetuity. 

In my opinion, the new UI is largely successful in improving the UX for filtering. Yours is the first negative feedback I've had on the change thus far, and what feedback I have had has been positive. 

> The whole change made the UX worse overall for the sake of looking good. Function should _always_ come before form.

_Every_ decision that I made regarding the filter rework was made in the interests of improving user experience.

**Comment by DingDongSoLong4** [comment]
For the vast majority people, I do believe that the new UI is an improvement on the old design. All your points are valid improvements which I agree with, but I definitely do not believe that they warrant completely reverting the change or introducing maintenance burden by having both UIs available.

The checkboxes are a simple improvement, and shouldn't be too difficult to add (I hope).
Adding a search box as in #3609, where it is selected by default, would in my opinion allow you to be just as fast as before.

#3549 is in the same vein as this issue, and has useful suggestions to improve the UI as opposed to just reverting it.

**Comment by DogmaDragon** [comment]
> <...> such as the performer gender filter; it uses checkboxes which implies you can select multiple

Addressed by https://github.com/stashapp/stash/pull/4419.

> When the filter options were in a select element, you could display more of them at once, you could arrow-key through them, and you could literally type out the filter to jump directly to it.

Filter search field https://github.com/stashapp/stash/pull/3740, pinned filters https://github.com/stashapp/stash/pull/3740 and sidebar https://github.com/stashapp/stash/pull/6547 should largely address any previous regressions.
